sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -49,6 +49,11 @@ struct HasArrowOperator<T,
                         std::void_t<decltype(std::declval<T>().operator->())>>
     : std::true_type {};
 
+// Checks that T is a reference to either device_global or
```

Comment:
I'd prefer to add `device_global` forward declaration and fully define `IsDeviceGlobalOrBaseRef` here.

Reply:
Sure!